docs/en/development/building-forms.md

```diff
@@ -0,0 +1,110 @@
+  const { initialConfiguration, update, isReadOnly, ...formProps } = useConfigLink<GlobalConfigurationDto>(link);
+  const { formState, handleSubmit, register, reset, control } = useForm<GlobalConfigurationDto>({
+    // mode onChange should be specified so that validation takes place immediately!
+    mode: "onChange",
+  });
+
+  // ...
+};
+```
+
+ConfigurationForm only takes care of the display of the component. All the logic now lives in the hook.
+
+Registering your own `onChange`-handler is not necessary anymore.
+`onSubmit` `handleSubmit`-function passes own submit function, which is called with filled form data type.
+
+In the `register`-function you can specify additional options for validation.
+For example, _required, min, max, pattern_.
+
+```tsx
+return (
+  <ConfigurationForm isValid={formState.isValid} isReadOnly={isReadOnly} onSubmit={handleSubmit(update)} {...formProps}>
+    <Title title={t("settings.title")} />
+    <Checkbox
+      label={t("fastForwardOnly.label")}
+      helpText={t("fastForwardOnly.helpText")}
+      disabled={isReadOnly}
+      {...register("fastForwardOnly", { shouldUnregister: true })}
+    />
+    <InputField
+      label={t("branchesAndTagsPatterns.label")}
+      helpText={t("branchesAndTagsPatterns.helpText")}
+      disabled={isReadOnly}
+      {...register("branchesAndTagsPatterns")}
+    />
+    <GpgVerificationControl control={control} isReadonly={isReadOnly} />
+  </ConfigurationForm>
+);
+```
+
+#### Note when using `formState`
+
+Be sure to use as proxy to get objects out (not formState.isValid!), because you won't notice the render cycle otherwise.
+
+#### Set to initial values
+
+In synchronous loading, a form can be set to an initial value using `defaultValue`.
+In the asynchronous case, values for each field can be set separately by using `defaultValue={stored.fastForwardOnly}` or an entire form using `reset`.
+
+```tsx
+useEffect(() => {
+  if (initialConfiguration) {
+    reset(initialConfiguration);
+  }
+}, [initialConfiguration]);
+```
+
+### Note when Creating new Components
+
+- If possible, pass all props.
+- React Hook Form needs the following values for event to be recognized: name, onChange, onBlur, ref (reference to input element).
+- `FormFieldTypes` is not a base, but helps for backwards compatibility with old function types. When writing a new component omit old onChange!
+- Since some components have other elements built around an input field, there is also the `forwardRef`. It creates a reference that can be passed to an inner element.
+- Nested forms are a bit more complex to build and might need a wrapper.
+- Validation rules are all based on the HTML standard and also allow for custom validation methods.
+- Fields marked as `disabled` in SCM-Manager cannot be disabled as `disabled={true}`, because then the object contained in onSubmit will also no longer contain the field (undefined). `readOnly` is the better choice.
```

Comment:
```suggestion
- Fields marked as `disabled` in SCM-Manager wont be included on submission. If you want to prevent interaction but need to submit the value of a form element, `readOnly` is the better choice.
```